src/js/containers/search/filters/psc/PSCListContainer.jsx

```diff
@@ -0,0 +1,176 @@
+import { Search } from 'js-search';
+import PropTypes from 'prop-types';
+
+import * as SearchHelper from 'helpers/searchHelper';
+import * as autocompleteActions from 'redux/actions/search/autocompleteActions';
+
+import Autocomplete from 'components/sharedComponents/autocomplete/Autocomplete';
+
+const propTypes = {
+    selectPSC: PropTypes.func,
+    setAutocompletePSC: PropTypes.func,
+    selectedPSC: PropTypes.object,
+    autocompletePSC: PropTypes.array
+};
+
+export class PSCListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            pscSearchString: '',
+            autocompletePSC: [],
+            noResults: false
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.clearAutocompleteSuggestions = this.clearAutocompleteSuggestions.bind(this);
+        this.timeout = null;
+    }
+
+    componentDidMount() {
+        this.parseAutocompletePSC(this.props.autocompletePSC);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!isEqual(nextProps.autocompletePSC, this.props.autocompletePSC)) {
+            this.parseAutocompletePSC(nextProps.autocompletePSC);
+        }
+    }
+
+    parseAutocompletePSC(psc) {
+        const values = [];
+        if (psc && psc.length > 0) {
+            psc.forEach((item) => {
+                values.push({
+                    title: item.product_or_service_code,
+                    subtitle: '',
+                    data: item
+                });
+            });
+        }
+
+        this.setState({
+            autocompletePSC: values
+        });
+    }
+
+    queryAutocompletePSC(input) {
+        this.setState({
+            noResults: false
+        });
+
+        // Only search if input is 2 or more characters
+        if (input.length >= 2) {
+            this.setState({
+                pscSearchString: input
+            });
+
+            if (this.pscSearchRequest) {
+                // A request is currently in-flight, cancel it
+                this.pscSearchRequest.cancel();
+            }
+
+            const pscSearchParams = {
+                search_text: this.state.pscSearchString
+            };
+
+            this.pscSearchRequest = SearchHelper.fetchPSC(pscSearchParams);
+
+            this.pscSearchRequest.promise
+                .then((res) => {
+                    const data = res.data.results;
+                    let autocompleteData = [];
+                    const search = new Search('product_or_service_code');
+                    search.addIndex('product_or_service_code');
+                    search.addDocuments(data);
+                    const results = search.search(this.state.pscSearchString);
+                    let improvedResults = slice(results, 0, 10);
+
+                    // Remove 'identifier' from selected PSC to enable comparison
+                    improvedResults = this.props.selectedPSC.toArray()
```

Comment:
It's not my library, take a look at the docs and the [library source](https://github.com/bvaughn/js-search/tree/master/source) to check.